test/interface/event-target.js

```diff
@@ -0,0 +1,94 @@
+const assert = require('../assert.js').for('EventTarget');
+
+const { parseHTML } = global[Symbol.for('linkedom')];
+
+const { Event, document, EventTarget } = parseHTML(
+  '<html><div id="container"><button id="buttonTarget" type="button">Click me!</button></div></html>',
+);
+
+// check basics
+
+let callCount = 0;
+const basicHandler = () => {
+  callCount++;
+};
+
+
+const eventTarget = new EventTarget();
+eventTarget.addEventListener('foo', basicHandler);
+eventTarget.dispatchEvent(new Event('foo'));
+assert(callCount, 1, 'basicHandler should have been called');
+
+assert(
+  eventTarget.dispatchEvent(new Event('click')),
+  true,
+  'Dispatching an event type with no handlers should return true',
+);
+assert(callCount, 1, 'Dispatching an event type should only call appropriate listeners');
+
+eventTarget.removeEventListener('foo', basicHandler);
+eventTarget.dispatchEvent(new Event('foo'));
+assert(callCount, 1, 'basicHandler should not have been called after being removed');
+
+assert(eventTarget._getParent(), null, 'getParent should return null');
+
+
+// check propagation now
+callCount = 0;
+const buttonTarget = document.getElementById('buttonTarget');
+const containerTarget = document.getElementById('container');
+const bodyTarget = document;
+buttonTarget.addEventListener('click', basicHandler, { once: true });
+containerTarget.addEventListener('click', basicHandler, { once: true });
+bodyTarget.addEventListener('click', basicHandler, { once: true });
+
+buttonTarget.dispatchEvent(new Event('click', { bubbles: true }));
+assert(callCount, 3, 'Event bubbling, listener should be called 3 times');
+
+
+// ensure once removed listeners
+buttonTarget.dispatchEvent(new Event('click', { bubbles: true }));
+assert(callCount, 3, 'listeners should only have been called once then removed');
+
+// check no bubbling
+callCount = 0;
+buttonTarget.addEventListener('click', basicHandler, { once: true });
+containerTarget.addEventListener('click', basicHandler, { once: true });
+bodyTarget.addEventListener('click', basicHandler, { once: true });
+
+buttonTarget.dispatchEvent(new Event('click', { bubbles: false }));
+assert(callCount, 1, 'Expect listener to be called once');
+
+// check stop propagation
+buttonTarget.addEventListener(
+  'click',
+  (event) => {
+    event.stopPropagation();
+    callCount++;
+  },
+  {
+    once: true,
+  },
+);
+containerTarget.addEventListener('click', basicHandler, { once: true });
+
+callCount = 0;
+buttonTarget.dispatchEvent(new Event('click', { bubbles: true }));
+assert(callCount, 1, 'listener should be called once before stopping bubbling');
+
+// check stop immediate propagation
+buttonTarget.addEventListener(
+  'click',
+  (event) => {
+    event.stopImmediatePropagation();
+    callCount++;
+  },
+  {
+    once: true,
+  },
+);
+containerTarget.addEventListener('click', basicHandler, { once: true });
```

Comment:
Testing `stopImmediatePropagation` there.
Once EventTarget support that feature I'll add more event handlers to confirm the full behaviour.

Reply:
sorry it took a while, but I've published `@ungap/event-target` and now everything "*should*" 🤞 come together

Reply:
I saw that, thanks! I thought it was quite fast tbh. I'll add more tests probably tomorrow.

As a side note I created a PR in node js about `stopImmediatePropagation`:
https://github.com/nodejs/node/pull/39463

Reply:
Nice one!

Reply:
test updated, now add 3 listeners to the button and one to the parent, check that `stopImmediatePropagation` stop running listeners at the right moment.